data_model/src/lib.rs

```diff
@@ -507,6 +510,30 @@ pub enum Value {
     BlockHeader(BlockHeaderValue),
 }
 
+/// Trait to identify [`ValueKind`] of a type which can be converted to a [`Value`].
+///
+/// Currently used only for permission tokens, so there is no need to implement it
+/// for every type convertible to [`Value`] until some permission token needs this.
+///
+/// Will be removed as well as builtin permission tokens and validators
+/// when *runtime validators* and *runtime permissions* will be properly implemented.
+pub trait CorrespondingValueKind: Into<Value> {
+    /// Kind of the [`Value`] which the implementing type can be converted to.
+    const KIND: ValueKind;
+}
+
+impl CorrespondingValueKind for u32 {
+    const KIND: ValueKind = ValueKind::U32;
+}
+
+impl CorrespondingValueKind for u128 {
+    const KIND: ValueKind = ValueKind::U128;
+}
+
+impl<I: Into<IdBox> + Into<Value>> CorrespondingValueKind for I {
+    const KIND: ValueKind = ValueKind::Id;
+}
+
```

Comment:
This needs a much better name. `PermissionTokenValueTrait` sounds a lot more descriptive of what it is.

Reply:
Ok. I'll move it to `data_model/src/permissions.rs` then